starlette/testclient.py

```diff
@@ -85,6 +85,14 @@ class WebSocketDenialResponse(  # type: ignore[misc]
     """
 
 
+class _Eof(enum.Enum):
+    EOF = enum.auto()
+
+
+EOF: typing.Final = _Eof.EOF
+Eof = typing.Literal[_Eof.EOF]
```

Comment:
The problem this solves is not related to this PR then?

Reply:
yes probably best to split into two PRs and add a changelog entry for the lost shutdown exception issue